compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -2084,7 +2084,7 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
             ExprType(info1.resType)
           case info1 => info1
 
-        isSubInfo(info1, info2, m.symbol.info.orElse(info1))
+        (isSubInfo(info1, info2, m.symbol.info.orElse(info1)) && m.symbol.hasTargetName(m.symbol.name))
         || matchAbstractTypeMember(m.info)
         || (tp1.isStable && m.symbol.isStableMember && isSubType(TermRef(tp1, m.symbol), tp2.refinedInfo))
```

Comment:
Why add the target name test only in the first of the three alternatives, and not in all of them? Couldn't we have the same issue in other cases?